apis/duck/v1/knative_reference.go

```diff
@@ -128,3 +128,8 @@ func isKReferenceGroupAllowed(ctx context.Context) bool {
 func KReferenceGroupAllowed(ctx context.Context) context.Context {
 	return context.WithValue(ctx, isGroupAllowed{}, struct{}{})
 }
+
+func (kr *KReference) String() string {
+	return fmt.Sprintf("Kind = %s, Namespace = %s, Name = %s, APIVersion = %s, Group = %s, Address = %s",
+		kr.Kind, kr.Namespace, kr.Name, kr.APIVersion, kr.Group, *kr.Address)
```

Comment:
`kr.Address` might be nil, so it can panic
```suggestion
        address := ""
        if kr.Address != nil {
        	address = *kr.Address
        }
	return fmt.Sprintf("Kind = %s, Namespace = %s, Name = %s, APIVersion = %s, Group = %s, Address = %s",
		kr.Kind, kr.Namespace, kr.Name, kr.APIVersion, kr.Group, address)
```